### PR TITLE
Use chrono data types for type-safety and readability where possible

### PR DESCRIPTION
**Pull request by Ashod**
### Summary

This is an attempt at avoiding mixing int/double with implicit units of time in favor of std::chrono, which does all that and more in a type safe and efficient fashion.

### Checklist

- [x] Code is properly formatted
- [x] All commits have Change-Id
- [x] I have run tests with `make check`
- [x] I have issued `make run` and manually verified that everything looks okay
- [x] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: kit/ChildSession.cpp [hunk 1]
```diff
@@ -733,9 +733,9 @@ bool ChildSession::sendFontRendering(const char* /*buffer*/, int /*length*/, con
 
     ptrFont = getLOKitDocument()->renderFont(decodedFont.c_str(), decodedChar.c_str(), &width, &height);
 
-    const auto duration = std::chrono::system_clock::now() - start;
-    const auto elapsed = std::chrono::duration_cast<std::chrono::microseconds>(duration).count();
-    LOG_TRC("renderFont [" << font << "] rendered in " << elapsed << "ms");
```
Comment (Ashod) [inline kit/ChildSession.cpp:738]: Apparently we were reporting microseconds as milliseconds, giving 1000x longer rendering time.

File: kit/Kit.cpp [hunk 2]
```diff
@@ -1192,15 +1192,15 @@ class Document final : public DocumentManagerInterface
...
             LOG_DBG("Calling lokit::documentLoad(" << FileUtil::anonymizeUrl(pURL) << ", \"" << options << "\").");
-            const auto start = std::chrono::system_clock::now();
+            const auto start = std::chrono::steady_clock::now();
             _loKitDocument.reset(_loKit->documentLoad(pURL, options.c_str()));
 #ifdef __ANDROID__
             _loKitDocumentForAndroidOnly = _loKitDocument;
 #endif
-            const auto duration = std::chrono::system_clock::now() - start;
-            const auto elapsed = std::chrono::duration_cast<std::chrono::microseconds>(duration).count();
-            const double totalTime = elapsed/1000.;
```
Comment (Ashod) [inline kit/Kit.cpp:1202]: Not sure why we got duration in microseconds only to divide into milliseconds.

File: wsd/TileCache.cpp [hunk 3]
```diff
@@ -138,17 +145,6 @@ void TileCache::forgetTileBeingRendered(const std::shared_ptr<TileCache::TileBei
     _tilesBeingRendered.erase(tileBeingRendered->getTile());
 }
 
-double TileCache::getTileBeingRenderedElapsedTimeMs(const TileDesc &tileDesc) const
```
Comment (Ashod) [inline wsd/TileCache.cpp:141]: Apparently unused.

**Review by tml1024 — APPROVED** [review]

**Comment by mmeeks** [comment]
Looks good to me with a fairly careful read - OTOH it is a big and perhaps risky change that will inevitably overlap with other changes; so we'll want to back-port it to cp-6.4 I guess (?) which will need a chunk of testing.

**Pull request merged**